src/print.rs

```diff
@@ -16,6 +16,43 @@ use crate::module::ALL_MODULES;
 use crate::modules;
 use crate::segment::Segment;
 
+pub struct Grapheme<'a>(&'a str);
+
+impl<'a> Grapheme<'a> {
+    pub fn width(&self) -> usize {
+        self.0.chars().flat_map(|c| c.width()).max().unwrap_or(0)
+    }
+}
+
+pub trait UnicodeWidthGraphemes {
+    fn width_graphemes(&self, is_extended: bool) -> usize;
+}
+
+impl UnicodeWidthGraphemes for str {
+    #[inline]
+    fn width_graphemes(&self, is_extended: bool) -> usize {
+        self.graphemes(is_extended)
+            .map(Grapheme)
+            .map(|g| g.width())
+            .sum()
+    }
+}
+
+impl UnicodeWidthGraphemes for String {
```

Comment:
If you implement `UnicodeWidthGraphemes` for `AsRef<str>` instead, you should be able to get rid of the extra `impl` here.

Reply:
I was looking for that! Thank you!